src/main/java/hudson/plugins/promoted_builds/integrations/jobdsl/PromotionContext.java

```diff
@@ -1,108 +1,185 @@
 package hudson.plugins.promoted_builds.integrations.jobdsl;
 
 import groovy.lang.Closure;
+import groovy.lang.MetaClass;
 import groovy.util.Node;
+import groovy.util.NodeBuilder;
+import groovy.util.NodeList;
+import groovy.util.XmlParser;
 import hudson.plugins.promoted_builds.PromotionCondition;
 import hudson.plugins.promoted_builds.conditions.DownstreamPassCondition;
 import hudson.plugins.promoted_builds.conditions.ParameterizedSelfPromotionCondition;
 import hudson.plugins.promoted_builds.conditions.SelfPromotionCondition;
 import hudson.plugins.promoted_builds.conditions.UpstreamPromotionCondition;
+import javaposse.jobdsl.dsl.DslContext;
+import javaposse.jobdsl.dsl.Item;
+import javaposse.jobdsl.dsl.JobManagement;
+import javaposse.jobdsl.dsl.helpers.step.StepContext;
+import javaposse.jobdsl.plugin.DslEnvironment;
+import org.codehaus.groovy.runtime.InvokerHelper;
+import org.codehaus.groovy.runtime.MethodClosure;
 
 import java.util.ArrayList;
 import java.util.List;
 
-import javaposse.jobdsl.dsl.Context;
-import javaposse.jobdsl.dsl.helpers.step.StepContext;
+import static javaposse.jobdsl.plugin.ContextExtensionPoint.executeInContext;
 
-import javaposse.jobdsl.plugin.DslEnvironment;
+class PromotionContext extends Item {
 
-import static javaposse.jobdsl.plugin.ContextExtensionPoint.executeInContext;
+    private static class ConfigureConditions {
+        public static final String METHOD_NAME = "configureConditions";
+        private final List<Node> conditions;
+        public ConfigureConditions(List<Node> conditions) {
+            this.conditions = conditions;
+        }
+        public void configureConditions(Node promotion) {
+            for (final Node condition : conditions) {
+                ((Node) ((NodeList) promotion.get("conditions")).get(0)).append(condition);
+            }
+        }
+    }
 
-class PromotionContext implements Context {
-    private final DslEnvironment dslEnvironment;
+    /**
+     * @deprecated Use automatically generated DSL for complete coverage of available {@link PromotionCondition}
+     * implementations
+     */
+    @Deprecated
+    private static class ConfigureLegacyConditions {
```

Comment:
This class was introduced as deprecated as a migration path to maintain JobDSL scripts compatibility with the existing manually defined DSL but to remind developers this approach should not be favored because of related issues.

A `@deprecated` description will be added unless another approach is recommended